src/v/utils/retry_chain_node.cc

```diff
@@ -300,6 +300,10 @@ basic_retry_chain_node<Clock>::get_poll_interval() const {
 
 template<class Clock>
 typename Clock::duration basic_retry_chain_node<Clock>::get_timeout() const {
+    vassert(
```

Comment:
i am wondering if this can be a `static_assert`, this way we would be warned at compile time

Reply:
i don't think the deadline is generally known at compile time. am i missing something?

Reply:
i don't really think this is a good place for the assertion anyway. may just move it to `client_pool`

Reply:
ditched all of this and put a conditional guard in client pool to avoid hitting overflows in watchdog.